index.js

```diff
@@ -17,7 +17,7 @@ function parse (text, reviver, options) {
   const protoAction = options.protoAction || 'error'
   const constructorAction = options.constructorAction || 'error'
 
-  if (Buffer.isBuffer(text)) {
+  if (typeof Buffer !== 'undefined' && Buffer.isBuffer(text)) {
```

Comment:
Should the feature detection occur once, outside of the parse function?

```js
const hasBuffer = typeof Buffer !== 'undefined';
```

Reply:
I think this is the kind of microoptimization where the performance is very hard to correctly assess without measuring, but I’m happy to move up if it gives the team a peace of mind.

Reply:
That'd be great, thanks!

Reply:
I’ve added this to the PR, but perhaps you want to reconsider. I have noticed there is already a benchmarking setup in the project and used that to compare the two implementations. From a cursory glance there is no meaningful difference between the two options.

<details><summary>Benchmark output</summary>

```
JSON.parse x 589,128 ops/sec ±0.71% (82 runs sampled)
secure-json-parse x 287,585 ops/sec ±0.59% (85 runs sampled)
cached x 285,247 ops/sec ±0.98% (84 runs sampled)

JSON.parse x 583,609 ops/sec ±0.65% (82 runs sampled)
secure-json-parse x 286,317 ops/sec ±0.73% (85 runs sampled)
cached x 288,569 ops/sec ±0.80% (86 runs sampled)

JSON.parse x 589,242 ops/sec ±0.76% (84 runs sampled)
secure-json-parse x 286,993 ops/sec ±0.65% (84 runs sampled)
cached x 284,582 ops/sec ±0.66% (83 runs sampled)

JSON.parse x 582,217 ops/sec ±0.79% (84 runs sampled)
secure-json-parse x 283,031 ops/sec ±0.81% (86 runs sampled)
cached x 284,793 ops/sec ±0.69% (87 runs sampled)
```

</details>

Theoretically the caching does change the behavior, but I am hard pressed to come up with a scenario where that difference will be obeserved.

Reply:
@denis-sokolov There is literally no downside to doing it the slightly better way, though.

Reply:
I don’t mind, I’ve pushed the commit as per your preferences.

Do note that you’re begging the question when you assume that it’s the better way. The performance could go either way, the timing is more correct at the time of use in the (way unlikely) event of Buffer dynamically showing in disappearing, the memory use is (laughably) smaller without an additional variable. What makes caching better?

Reply:
You are right. I retract my statement of it being strictly better. It really depends in this case.

Reply:
Thanks for agreeing. I’m leaving the cached version in for now, but I’m happy to drop it if anyone decides that’s preferred.